java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -196,20 +186,21 @@ public boolean equals(Object o) {
     return Objects.equals(values, that.values) &&
       Objects.equals(constraints, that.constraints) &&
       Objects.equals(exitSymbolicValue, that.exitSymbolicValue) &&
-      Objects.equals(stack, that.stack);
+      Objects.equals(stack, that.stack) &&
+      Objects.equals(lastAssociatedSymbol, that.lastAssociatedSymbol);
```

Comment:
Why do you need it in the equality? `lastEvaluated` symbol was not used for it, and I fear that it will destroy caching of PS.